tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py

```diff
@@ -490,4 +524,80 @@ def test_numeric_decimal_incompatible(self):
         conversion = duckdb.query_df(x, "x", "select * from x").fetchall()
 
         assert(conversion == reference)
+        print(reference)
+        print(conversion)
+
+    #result: [('1E-28',), ('10000000000000000000000000.0',)]
+    def test_numeric_decimal_combined(self):
+        duckdb_conn = duckdb.connect()
+        decimals = pd.DataFrame(
+            data={
+                "0": [
+                    Decimal("0.0000000000000000000000000001"),
+                    Decimal("10000000000000000000000000.0")
+                ]
+            }
+        )
+        reference_query = """
+            CREATE TABLE tbl AS SELECT * FROM (
+                VALUES
+                (0.0000000000000000000000000001),
+                (10000000000000000000000000.0),
+            ) tbl(a);
+        """
+        duckdb_conn.execute(reference_query)
+        reference = duckdb.query("select * from tbl", connection=duckdb_conn).fetchall()
+        conversion = duckdb.query_df(decimals, "x", "select * from x").fetchall()
+        assert(conversion == reference)
+        print(reference)
+        print(conversion)
+
+    #result: [('1234.0',), ('123456789.0',), ('1234567890123456789.0',), ('0.1234567890123456789',)]
+    def test_numeric_decimal_varying_sizes(self):
+        duckdb_conn = duckdb.connect()
+        decimals = pd.DataFrame(
+            data={
+                "0": [
+                    Decimal("1234.0"),
+                    Decimal("123456789.0"),
+                    Decimal("1234567890123456789.0"),
+                    Decimal("0.1234567890123456789")
+                ]
+            }
+        )
+        reference_query = """
+            CREATE TABLE tbl AS SELECT * FROM (
+                VALUES
+                    (1234.0),
+                    (123456789.0),
+                    (1234567890123456789.0),
+                    (0.1234567890123456789)
+            ) tbl(a);
+        """
+        duckdb_conn.execute(reference_query)
+        reference = duckdb.query("select * from tbl", connection=duckdb_conn).fetchall()
+        conversion = duckdb.query_df(decimals, "x", "select * from x").fetchall()
+        assert(conversion == reference)
+        print(reference)
+        print(conversion)
+
+    def test_numeric_decimal_out_of_range(self):
+        duckdb_conn = duckdb.connect()
+        data = [Decimal("1.234567890123456789012345678901234567"), Decimal("123456789012345678901234567890123456.0")]
```

Comment:
Python decimals can be arbitrarily large, do we have tests with decimals larger than we can handle somewhere? (e.g. `decimal.Decimal('123456789123456789123456789123456789123456789.5')`)?

Reply:
Ah that's true, I dont think we do.
The example you sent had more than 38 digits so I truncated it to test the Decimal.
It fell back to VARCHAR, which should indeed be DOUBLE instead.
I'll add a test and the functionality first thing when I come in today :+1: